hack/heap/hh_shared.c

```diff
@@ -148,7 +148,7 @@
  * with the MAP_ANONYMOUS flag. The memfd_create() system call first
  * appeared in Linux 3.17.
  ****************************************************************************/
-#if !defined __APPLE__ && !defined _WIN32
+#ifdef __linux__
```

Comment:
Reading back on the penultimate comment it may not have been clear, but what I was getting at was preferring `#ifndef __linux__` over `#ifdef __APPLE__` as a test, since generally the BSDs are close enough that you can cover them all together in the `#ifndef __linux__` branch.